### PR TITLE
fix(app): resolve module location conflicts through deck config during protocol setup on ODD

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Resolve location conflicts in the on device display's protocol setup flow by updating deck
configuration accordingly.

Closes [PLAT-287](https://opentrons.atlassian.net/browse/PLAT-287), PLAT-291

# Review requests

- resolve location conflicts during the deck hardware setup step of protocol run setup on the ODD 

# Risk assessment
low

[PLAT-287]: https://opentrons.atlassian.net/browse/PLAT-287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mjhuff — COMMENTED** [review]
Thanks for the global `useNotifyService` mock!

I think fixing the failing test is hopefully 🤞  as simple as a `vi.unmock` in `useNotifyService.test.ts`

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/ChooseModuleToConfigureModal.tsx [hunk 1]
```diff
@@ -50,9 +58,15 @@ export const ChooseModuleToConfigureModal = (
...
     requiredModuleModel,
     isOnDevice,
+    robotName,
+    displaySlotName,
   } = props
   const { t } = useTranslation(['protocol_setup', 'shared'])
-  const attachedModules = useModulesQuery().data?.data ?? []
+  const history = useHistory()
+  const currentRunId = useCurrentRunId()
+  const { stopRun } = useStopRunMutation()
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/ChooseModuleToConfigureModal.tsx]: should this be `useDismissCurrentRunMutation`? stop run doesn't navigate away from the run setup/modal

**Review by brenthagen — APPROVED** [review]
some ui nits, but the substantive conflict changes look good to me !

Screen Shot 2024-04-23 at 1 52 44 PM
Screen Shot 2024-04-23 at 1 53 29 PM

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/ChooseModuleToConfigureModal.tsx (hunk 1 above)
Comment (b-cooper) [reply app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/ChooseModuleToConfigureModal.tsx]: Oh good call, we may actually have to do both. I'll look into dismissing an un-started idle run.

**Pull request merged**